### PR TITLE
FIX: connect uses append on the upstreams, we need to make sure that

### PR DESCRIPTION
**Pull request by CJ-Wright**
the upstreams is appendable

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@mrocklin ready for review

**Comment by martindurant** [comment]
I can't see this causing any problem, but I also can't see a situation in which it's necessary. @CJ-Wright , did you come across a problem, perhaps something we should test for?

**Comment by CJ-Wright** [comment]
@martindurant please see the test

File: streamz/tests/test_core.py [hunk 1]
```diff
@@ -696,6 +696,28 @@ def test_connect():
     assert L == [3, 5]
 
 
+def test_multi_connect():
+    source0 = Stream()
+    source1 = Stream()
+    source_downstream = source0.union(source1)
+    # connect assumes this default behaviour
+    # of stream initialization
+    assert not(source_downstream.downstreams)
```
Comment (martindurant) [inline streamz/tests/test_core.py]: should be `not source_downstream.downstreams` ?

**Comment by CJ-Wright** [comment]
The reason why this fails is because `self.upstreams` was stored as a tuple, and `Stream.connect` tries to append to that tuple, which is invalid.

**Comment by martindurant** [comment]
The test failure appears unrelated?

**Comment by CJ-Wright** [comment]
I think so, rerunning to see if it reproduces (it didn't on my local)

**Comment by martindurant** [comment]
Should merge from master following #160

**Pull request merged**